cmd/syncthing/main.go

```diff
@@ -372,6 +379,37 @@ func main() {
 	}
 }
 
+func upgradeViaRest() error {
+	var cfg config.Configuration
+	fd, err := os.Open(locations[locConfigFile])
+	if err != nil {
+		return err
+	}
+	defer fd.Close()
+	err = xml.NewDecoder(fd).Decode(&cfg)
+	if err != nil {
+		l.Fatalln("Cannot upgrade, database seems to be locked and the config cannot be found")
+	}
+	target := cfg.GUI.Address
+	// TODO Can a target contain *:// ?
+	if cfg.GUI.UseTLS {
+		target = "https://" + target
+	} else {
+		target = "http://" + target
+	}
+	r, _ := http.NewRequest("POST", target+"/rest/system/upgrade", nil)
+	r.Header.Set("X-API-Key", cfg.GUI.APIKey)
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // TODO Check the key
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   60 * time.Second,
+	}
+	_, err = client.Do(r)
+	return err
```

Comment:
Actually might want to get the response as well here and check the StatusCode for something != 200, i.e. syncthing will return 500 in case upgrades are disabled or the upgrade server unreachable, etc.